data-prepper-api/src/main/java/org/opensearch/dataprepper/model/configuration/DataPrepperVersion.java

```diff
@@ -58,21 +58,20 @@ public Optional<Integer> getMinorVersion() {
     /**
      * Determines if the provided Data Prepper Version is compatible with this.
      *
-     * The initial implementation is a basic implementation that enforces equivalent versions, any shorthand format version
-     * compared with full format with equivalent major versions and equivalent major version but the comparing minor version
-     * is less than this minor version are compatible.
+     * The initial implementation is a basic implementation that will return true only when the comparing version is
+     * less than or equal to this version.
      *
      * @param o - the other DataPrepperVersion to compare with
      * @return return true if the versions are compatible, otherwise false
      * @since 2.1
      */
     public boolean compatibleWith(DataPrepperVersion o) {
 
-        if (this.majorVersion != o.getMajorVersion()) {
+        if (this.majorVersion < o.getMajorVersion()) {
```

Comment:
Does the logic to return true if `this.majorVersion > o.getMajorVersion()` make sense? If it's arbitrary comparison why would 1 < 2 return false/incompatible and 2 < 1 return true/compatible?

Reply:
We are trying to determine if `this` is compatible with the version defined in `o`. `this` should be backwards compatible and support older versions and return true if the version when `o` is smaller and false when `o` is greater.

Reply:
Is this assuming that this did not contain breaking changes to older versions?

Reply:
Yes, it is assuming. Its a best effort.